lib/mox.ex

```diff
@@ -685,8 +685,11 @@ defmodule Mox do
       :ok ->
         :ok
 
+      {:error, error} when is_binary(error) ->
+        raise ArgumentError, error
```

Comment:
Can we get rid of this clause by returning 

```elixir
{:error, %ArgumentError{message: "..."}}
```

from the functions below instead? That way, you can turn the `case` clause below into

```elixir
{:error, error} when is_exception(error) ->
  raise error
```

and remove this one 🙃

Reply:
Actually on second thought, this change is not really necessary is it? What makes this better then returning tuples around and then converting them to messages with a `expectation_error_to_message/2` helper?

Reply:
The main reason was to handle any additional error from `get_and_update`, since today `expectation_error_to_message` handles the ownership errors. I created a `raise_cannot_add_expectation!`  function and moved the error handling to it. https://github.com/dashbitco/mox/pull/158/commits/79a2363d05139f26aa2d753149b7502c05f28eb0